common/spring-cloud-stream-modules-common-configuration/src/main/java/org/springframework/cloud/stream/module/metrics/MetricProperties.java

```diff
@@ -66,11 +63,15 @@ public void setName(String name) {
 		this.name = name;
 	}
 
-	public Expression getNameExpression() {
-		return spelExpressionParser.parseExpression(nameExpression);
+	public LiteralExpression getNameExpression() {
+		return nameExpression;
 	}
 
-	public void setNameExpression(String nameExpression) {
+	public LiteralExpression getComputedNameExpression() {
```

Comment:
Can't be a LiteralExpression for the general case. Only if a literal `name` has been provided